blog/2020-10-08-debezium-community-stories-with-renato-mefi.adoc

```diff
@@ -0,0 +1,143 @@
+
+*What are your use cases for Debezium and CDC in your current project?*
+
+It's a long and enjoyable story (long in terms of the internet), we've been using Debezium since Q4 2018, it's been 2 years at the moment I'm writing those answers here.
+
+When I classify Debezium within our product, I say it is an architectural component, the idea behind this is to position it as a platform/infrastructure concern, in a way that it can reach multiple parts of the stack and services. I consider this abstraction of Debezium one of the key success factors it had in its adoption and growth within our platform, let me explain this better!
+
+Our first use case is likely to be one of the most common ones for CDC, the https://martinfowler.com/bliki/StranglerFigApplication.html[Strangler pattern], which for us came before Debezium; so let me tell this part of the story first: when I joined Usabilla (later acquired by SurveyMoney), there was already an effort to move our platform to a new architecture and the Strangler pattern was already there, when the first couple of services started to grow, their primary way to bring data out of legacy was to poll the database, and needless to say, this could go very wrong! Our legacy database is a MongoDB cluster, and since I was preoccupied with the polling approach I started to dig into possibilities, I was hoping to find something like a streaming api for it, but what I ended up encountering was the database changelog, "oplog" as it's called in Mongo!
+
+It came to my mind right away: "Oh, I could write something that queries the data from the Oplog and sends it to Kafka". So I checked with our in-house Senior SRE and MongoDB expert https://twitter.com/gwkunze[Gijs Kunze] who thought it could be a good idea; as a next step I went to talk to my colleague https://twitter.com/rdohms[Rafael Dohms], and we decided to do some extra Googling, and like that, we found Debezium! It was the perfect match to our needs and better than what we could have written by ourselves!
+
+Now back to our use case, what makes it an architecture component for us, is basically the approach, we abstracted and wrapped Debezium in a project called Legacy Data Syncer (LDS for us, because acronyms never get old). Although it might look simple to spin up a Kafka Connect with Debezium, running it production ready, monitoring multiple collections within the database, exposing metrics, doing transformations and more, is not such an easy task. So how does it work? Every time an engineering team needs to capture data from our legacy system, to start strangling a feature, they only have to do two things, open a pull request which literally adds one line to LDS, and create their Kafka consumer! 
+
+[.centered-image.responsive-image]
+====
+++++
+<img src="/images/debezium_community_stories_with_renato_mefi_lds.png" style="max-width:100%;" class="responsive-image">
```

Comment:
@renatomefi, could you perhaps provide that image with a larger font size/resolution? It's rather small right now.

Reply:
I'm going to add this image to carbon.now.sh and the outbox one as well!
This way we can even match the colors to debezium style :)

Reply:
Legacy data syncer: https://carbon.now.sh/?bg=rgba(171%2C%20184%2C%20195%2C%201)&t=lucario&wt=none&l=yaml&ds=true&dsyoff=20px&dsblur=68px&wc=true&wa=true&pv=56px&ph=56px&ln=false&fl=1&fm=Hack&fs=14px&lh=133%25&si=false&es=2x&wm=false&code=%250Aconnectors%253A%250A%2520%2520config%253A%250A%2520%2520%2509%2523%2520Add%2520a%2520new%2520line%2520to%2520monitor%2520a%2520new%2520collection%2520in%2520MongoDB%250A%2520%2520%2520%2520%2523%2520Template%253A%250A%2520%2520%2520%2520%2523%2520%2522Collection%2520name%2522%253A%2520%255B%2522Enable%2520raw%2520Oplog%2522%252C%2520%2522Enable%2520CDC%2520transformation%2522%255D%250A%2520%2520%2520%2520%250A%2520%2520%2520%2520collection_a%253A%2520%255B%2522op%2522%252C%2520%2522cdc%2522%255D%250A%2520%2520%2520%2520%2523%2520derived%2520topics%253A%2520%2560legacy-data-syncer.op.collection-a%2560%2520%2560legacy-data-syncer.cdc.collection-a%2560%250A%2520%2520%2520%2520%2523%2520connector%2520names%253A%2520%2560op-collection_a%2560%2520%2560cdc-collection_a%2560%250A%2520%2520%2520%2520%250A%2520%2520%2520%2520collection_b%253A%2520%255B%2522op%2522%252C%2520%2522cdc%2522%255D%250A%2520%2520%2520%2520collection_c%253A%2520%255B%2522op%2522%252C%2520%2522cdc%2522%255D